### PR TITLE
Add some DP problems

### PR DESCRIPTION
**Pull request by donaldong**
As mentioned in #1, I started tracking the problem types pretty late.

Many problems here are leaning towards the hard side. And my difficulty rating may not be accurate since I don't remember most of these problems 😅

### CODE REVIEW

File: dp.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Dynamic Programming
+===================
+Dynamic Programming (DP) is to plan the best solution dynamically by using the
+result of solved subproblems. When we are solving DP problems, we need to think
+about the recurrence relation: how is knowing the results of solved subproblems
+going to help.
+
+
+DP problems can be solved recursively or iteratively. The iterative solutoins
```
Comment (FernandoCaudillo10) [inline dp.md]:
```suggestion
DP problems can be solved recursively or iteratively. The iterative solution
```

File: dp.md [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+usually have a performance advantage, however, the overhead of the recursive
```
Comment (FernandoCaudillo10) [inline dp.md]:
```suggestion
usually has a performance advantage, however, the overhead of the recursive
```

**Review by FernandoCaudillo10 — CHANGES_REQUESTED** [review]
Just small grammar changes.

File: dp.md (hunk 2 above)
Comment (donaldong) [reply dp.md]: Hmm, maybe it should be `The iterative solutions have performance advantages`?

Comment (FernandoCaudillo10) [reply dp.md]: I figured it would be `The iterative solution has performance advantages`, but now that I reread it, it sounds good both ways.

**Review by FernandoCaudillo10 — APPROVED** [review]

**Pull request merged**